tests/e2e/tracer/handlers/same_function_name.py

```diff
@@ -0,0 +1,26 @@
+from uuid import uuid4
+
+from aws_lambda_powertools import Tracer
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+tracer = Tracer()
+
+
+class Comments:
+    @tracer.capture_method
+    def get_all(self):
+        return [{"id": f"{uuid4()}", "completed": False} for _ in range(5)]
+
+
+class Todos:
+    @tracer.capture_method
+    def get_all(self):
+        return [{"id": f"{uuid4()}", "completed": False} for _ in range(5)]
+
```

Comment:
Create an ABC so it's closer to what customers reported to us when they hit this issue.

Reply:
Looks great now :rocket: 
![image](https://user-images.githubusercontent.com/4295173/188700070-f6f1994a-82d3-4a7d-8c06-d35f94c51f3a.png)